community/bolt/src/main/java/org/neo4j/bolt/v1/runtime/internal/SessionStateMachine.java

```diff
@@ -84,19 +84,29 @@ public State init( SessionStateMachine ctx, String clientName, Map<String,Object
                         }
                         catch ( AuthenticationException e )
                         {
-                            return error( ctx, new Neo4jError( e.status(), e.getMessage(), e ) );
+                            ctx.error( new Neo4jError( e.status(), e.getMessage() ));
+                            return INIT_ERROR;
                         }
                         catch ( Throwable e )
                         {
-                            return error( ctx, e );
+                            ctx.error( new Neo4jError( Status.General.UnknownError, e.getMessage() ));
+                            return halt( ctx );
                         }
                     }
 
                     @Override
+                    public State interrupt( SessionStateMachine ctx )
+                    {
+                        // this means that you cannot interrupt init
+                        // this ensures that a user has to either finish init or finish noImpl
+                        return UNINITIALIZED;
+                    }
+
+                    @Override
                     protected State onNoImplementation( SessionStateMachine ctx, String command )
                     {
-                        ctx.error( new Neo4jError( Status.Request.Invalid, "No operations allowed until you send an " +
-                                "INIT message." ) );
+                        ctx.error( new Neo4jError( Status.Security.Forbidden, "No operations allowed until you send an " +
```

Comment:
This is maybe not a security error since this will be called when a client is misusing the protocol. Like trying to do stuff before calling INIT